src/ui/character_selection.rs

```diff
@@ -621,7 +621,7 @@ pub fn select_character_system(
     )>,
 ) {
     let keyboard_input =
-        keyboard_input.just_pressed(KeyCode::D) || keyboard_input.just_released(KeyCode::A);
+        keyboard_input.just_pressed(KeyCode::D) || keyboard_input.just_pressed(KeyCode::A);
```

Comment:
On second look, idk what this _should_ be. Maybe `just_released` is more consistent. 

```
yui@shivaji:~/workspace/thetawave (feature/dynamicBackgroundImageAssetForWasm)$ rg 'just_pressed' src/
src/ui/character_selection.rs
624:        keyboard_input.just_pressed(KeyCode::D) || keyboard_input.just_released(KeyCode::A);
631:                gamepad_input.just_pressed(GamepadButton {
634:                }) || gamepad_input.just_pressed(GamepadButton {
```
```
yui@shivaji:~/workspace/thetawave (feature/dynamicBackgroundImageAssetForWasm)$ rg 'just_released' src/
src/ui/character_selection.rs
526:    let keyboard_join_input = keyboard_input.just_released(KeyCode::ShiftLeft)
527:        || keyboard_input.just_released(KeyCode::ShiftRight);
559:                gamepad_input.just_released(GamepadButton {
624:        keyboard_input.just_pressed(KeyCode::D) || keyboard_input.just_released(KeyCode::A);

src/options/display.rs
47:    let fullscreen_input = keyboard_input.just_released(KeyCode::F);
74:    let zoom_input = keyboard_input.just_released(KeyCode::V);

src/states/pause_menu.rs
19:    let mut pause_input = keyboard_input.just_released(KeyCode::Escape);
22:        pause_input |= gamepad_input.just_released(GamepadButton {
65:    let mut unpause_input = keyboard_input.just_released(KeyCode::Escape);
68:        unpause_input |= gamepad_input.just_released(GamepadButton {

src/states/mod.rs
203:    let mut reset = keyboard_input.just_released(KeyCode::R);
206:        reset |= gamepad_input.just_released(GamepadButton {

src/states/game.rs
18:    let mut start_input = keyboard_input.just_released(KeyCode::Return)
19:        || keyboard_input.just_released(KeyCode::Space);
22:        start_input |= gamepad_input.just_released(GamepadButton {
59:    let mut start_input = keyboard_input.just_released(KeyCode::Return)
60:        || keyboard_input.just_released(KeyCode::Space);
63:        start_input |= gamepad_input.just_released(GamepadButton {
99:    let mut start_input = keyboard_input.just_released(KeyCode::Return)
100:        || keyboard_input.just_released(KeyCode::Space);
103:        start_input |= gamepad_input.just_released(GamepadButton {
139:    let mut quit_input = keyboard_input.just_released(KeyCode::Escape);
142:        quit_input |= gamepad_input.just_released(GamepadButton {
```

Reply:
I'd say consistency doesn't matter in other places, since both `just_pressed` & `just_released` have their use-cases:
- For things like menu navigation (or character selection), `just_pressed` feels more natural" since the update happens right away
- For toggles (confirmation, fullscreen, exit, back, etc), `just_released` feels more natural
My examples are not always the case, so it's nice to play around with both options & see what fits best.

Reply:
I like being on the front side of the beat for switching which character is highlighted/emphasized. I havent thought so much about when `just_released`  vs `just_pressed` is better but  that is a good summary now that I think about other games I have played :)